### PR TITLE
Add Auth header for batch request

### PR DESCRIPTION
**Pull request by Vorob-Astronaut**

### CODE REVIEW

**Comment by Vorob-Astronaut** [comment]
@googlebot I signed it!

**Comment by Vorob-Astronaut** [comment]
@googlebot I fixed

**Comment by Vorob-Astronaut** [comment]
@googlebot I fixed it.

File: lib/google/apis/core/batch.rb [hunk 1]
```diff
@@ -177,6 +177,8 @@ def build_head(call)
           call.header.each do |key, value|
             request_head << sprintf("\r\n%s: %s", key, value)
           end
+          token = call.options.authorization
+          request_head << "\r\nAuthorization: Bearer #{token}" unless token.nil?
```
Comment (blowmage) [inline lib/google/apis/core/batch.rb]: Partial solution, only works for a String authorization object.

**Comment by blowmage** [comment]
@Vorob-Astronaut thank you for your contribution! We will add to your contribution to ensure batch parts using Google Auth/Signet Client authorization objects also work. But it won't stop us from accepting your change once we get a green build.

**Pull request merged**